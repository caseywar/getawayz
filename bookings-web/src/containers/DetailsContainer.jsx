import React, { useState, useEffect } from 'react';
import { findPlaceById } from '../services/placesApi';
import PlaceDetails from '../components/places/PlaceDetails';

const DetailsContainer = ({ match }) => {
    const [place, setPlace] = useState({});
    const [loading, setLoading] = useState(true);

    useEffect(() => {
        findPlaceById(match.params.id)
        .then(setPlace)
        .finally(() => setLoading(false))
    }, []);

    if (loading) return <h1>Loading...</h1>

    return <PlaceDetails place={place} />
};

export default DetailsContainer;