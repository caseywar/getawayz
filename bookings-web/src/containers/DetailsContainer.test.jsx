require('dotenv').config();
import React from 'react';
import { setupServer } from 'msw/node';
import { rest } from 'msw';
import mockDetailsData from '../../json/detailsContainerJson.json';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import DetailsContainer from './DetailsContainer';
import { screen, waitFor } from '@testing-library/react';


// const server = setupServer(
//     rest.get('http://localhost:7890/api/v1/places/60ac55a713f85d2a6af9a1ff', (req, res, ctx) => {
//         console.log('server test')
//         return res(ctx.json(mockDetailsData));
//     })
// );

describe('Details Container', () => {
    // beforeAll(() => server.listen());
    // afterAll(() => server.close());

    it('it displays the place detail page', async () => {
        render(
            <MemoryRouter> 
                <DetailsContainer 
                    match={{
                        params: {
                            id: '60ac55a713f85d2a6af9a1ff'
                        }
                    }}
                />
                </MemoryRouter>
        );

        screen.getByText('Loading...');

        const div = screen.findByRole('div', { name: 'placeDetail' });

        return waitFor(() => {
            expect(div).toMatchSnapshot();
        })
    })
})