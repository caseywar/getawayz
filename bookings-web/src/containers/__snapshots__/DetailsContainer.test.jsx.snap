// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Container it displays the place detail page 1`] = `Promise {}`;
